internal/quaiapi/quai_api.go

```diff
@@ -693,14 +690,20 @@ func (s *PublicBlockChainQuaiAPI) fillSubordinateManifest(b *types.WorkObject) (
 }
 
 // ReceiveMinedHeader will run checks on the block and add to canonical chain if valid.
-func (s *PublicBlockChainQuaiAPI) ReceiveMinedHeader(ctx context.Context, raw json.RawMessage) error {
+func (s *PublicBlockChainQuaiAPI) ReceiveMinedHeader(ctx context.Context, raw hexutil.Bytes) error {
 	nodeCtx := s.b.NodeCtx()
-	// Decode header and transactions.
-	var woHeader *types.WorkObject
-	if err := json.Unmarshal(raw, &woHeader); err != nil {
+	protoWorkObject := &types.ProtoWorkObject{}
+	err := proto.Unmarshal(raw, protoWorkObject)
+	if err != nil {
 		return err
 	}
-	woHeader.Header().SetCoinbase(common.BytesToAddress(woHeader.Coinbase().Bytes(), s.b.NodeLocation()))
+
+	woHeader := &types.WorkObject{}
+	err = woHeader.ProtoDecode(protoWorkObject, s.b.NodeLocation(), types.PEtxObject)
+	if err != nil {
+		return err
+	}
+	woHeader.WorkObjectHeader().SetCoinbase(common.BytesToAddress(woHeader.Coinbase().Bytes(), s.b.NodeLocation()))
```

Comment:
This is probably unnecessary now because protodecode should give the right scope

Reply:
this one yeah, will delete this